packages/@aws-cdk/aws-appmesh/lib/mesh.ts

```diff
@@ -99,6 +110,62 @@ abstract class MeshBase extends cdk.Resource implements IMesh {
       mesh: this,
     });
   }
+
+  public grantRead(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:DescribeMesh',
+      'appmesh:ListMeshes',
+      'appmesh:DescribeVirtualService',
+      'appmesh:ListVirtualServices',
+      'appmesh:DescribeVirtualRouter',
+      'appmesh:ListVirtualRouter',
+      'appmesh:DescribeRoute',
+      'appmesh:ListRoutes',
+      'appmesh:DescribeVirtualNode',
+      'appmesh:ListVirtualNodes',
+      'appmesh:DescribeVirtualGateway',
+      'appmesh:ListVirtualGateways',
+      'appmesh:DescribeGatewayRoute',
+      'appmesh:ListGatewayRoutes',
+    );
+  }
+
+  public grantWrite(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:CreateMesh',
+      'appmesh:UpdateMesh',
+      'appmesh:DeleteMesh',
+      'appmesh:CreateVirtualNode',
+      'appmesh:UpdateVirtualNode',
+      'appmesh:DeleteVirtualNode',
+      'appmesh:CreateVirtualRouter',
+      'appmesh:UpdateVirtualRouter',
+      'appmesh:DeleteVirtualRouter',
+      'appmesh:CreateRoute',
+      'appmesh:UpdateRoute',
+      'appmesh:DeleteRoute',
+      'appmesh:CreateVirtualGateway',
+      'appmesh:UpdateVirtualGateway',
+      'appmesh:DeleteVirtualGateway',
+      'appmesh:CreateGatewayRoute',
+      'appmesh:UpdateGatewayRoute',
+      'appmesh:DeleteGatewayRoute',
+      'appmesh:CreateVirtualService',
+      'appmesh:UpdateVirtualService',
+      'appmesh:DeleteVirtualService',
+      'appmesh:TagResource',
+      'appmesh:UntagResource',
+    );
+  }
+
+  private grant(grantee: iam.IGrantable, ...actions: string[]) {
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [this.meshArn, this.meshArn + '/*'],
```

Comment:
I wonder - can we separate the permissions for these that are for the Mesh's ARN, and for those that require Mesh ARN + `'/*'`?

Reply:
Currently it looks like grants don't support multiple policy statements.

Reply:
How about we call `Grant.addToPrincipal()` twice - once with actions on the Mesh, and `resourceArns` equal to `[this.meshArn]`, and then a second time, with the sub-resource actions, and `resourceArns` equal to ``[`${this.meshArn}/*`]``, and return the result of that first call?